### PR TITLE
Bug Fix for Inspection Limit Feature for Application Gateway Firewall Policy Settings

### PR DESCRIPTION
**Pull request by Shawnli222**
## Description
Inspection Limit Feature for Application Gateway Firewall Policy Settings

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Review by tejasshah7 — APPROVED** [review]
LGTM, remember to create a WI to remove "Geo" in the next major release.

File: src/Network/Network/ChangeLog.md [hunk 1]
```diff
@@ -22,6 +22,8 @@
 * Added support for new Application Gateway SKU type, Basic SKU
 * Onboarded `Microsoft.EventGrid/partnerNamespaces` to private link cmdlets
 * Onboarded `Microsoft.EventGrid/namespaces` to private link cmdlets
+* Fixed bug in `NewAzureApplicationGatewayFirewallCustomRuleGroupByVariable` to add "GeoLocation" as a valid input for VariableName
+* Added breaking change message for parameter `VariableName` in `NewAzureApplicationGatewayFirewallCustomRuleGroupByVariable` tor remove "Geo" as a valid input.
```
Comment (tejasshah7) [inline src/Network/Network/ChangeLog.md:26]: Typo "tor".

Also, we are not introducing a breaking change anymore correct?

Comment (Shawnli222) [reply src/Network/Network/ChangeLog.md]: yes, this is only a message indicating that it is happening in Nov, I was asked to add this

**Review by tejasshah7 — APPROVED** [review]

File: src/Network/Network/FirewallPolicy/FirewallCustomRule/GroupByUserSession/GroupByVariable/NewAzureApplicationGatewayFirewallCustomRuleGroupByVariableCommand.cs [hunk 2]
```diff
@@ -21,11 +22,12 @@ namespace Microsoft.Azure.Commands.Network
     [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ApplicationGatewayFirewallCustomRuleGroupByVariable"), OutputType(typeof(PSApplicationGatewayFirewallCustomRuleGroupByVariable))]
     public class NewAzureApplicationGatewayFirewallCustomRuleGroupByVariableCommand : NetworkBaseCmdlet
     {
+        [CmdletParameterBreakingChangeWithVersionAttribute("VariableName", "11.0.0", "6.0.0", ChangeDescription = "Geo would be invalid for parameter VariableName")]
```
Comment (YanaXu) [inline src/Network/Network/FirewallPolicy/FirewallCustomRule/GroupByUserSession/GroupByVariable/NewAzureApplicationGatewayFirewallCustomRuleGroupByVariableCommand.cs:25]: The current Network version is 6.1.1. Thus, the 3rd parameter should be 7.0.0 (it means this parameter will be removed in 7.0.0).

Comment (Shawnli222) [reply src/Network/Network/FirewallPolicy/FirewallCustomRule/GroupByUserSession/GroupByVariable/NewAzureApplicationGatewayFirewallCustomRuleGroupByVariableCommand.cs]: fixed

**Review by YanaXu — APPROVED** [review]

**Pull request merged**